frontend/src/types/oauth.ts

```diff
@@ -25,15 +58,17 @@ export function redirectUrl(): string {
   return `${window.location.origin}/oauth/callback`;
 }
 
-type OAuthType = "login" | "register";
+// login: users try to login via oauth
+// register: users try to bind a vcs to her workspace
+export type OAuthType = "login" | "register-vcs" | "link-vcs-repository";
 
 export function openWindowForOAuth(
   endpoint: string,
   applicationId: string,
   type: OAuthType
 ): Window | null {
   // we use type to determine oauth type when receiving the callback
-  const stateQueryParameter = `${type}-${randomString(20)}`;
+  const stateQueryParameter = `${type}&${randomString(20)}`;
```

Comment:
& is a special character and we should avoid using it.

I think it's OK to use - as the separator, when extracting, we just trim the last component and get the type